nameko/containers.py

```diff
@@ -127,9 +127,25 @@ def __init__(self, service_cls, worker_ctx_cls, config):
         self.max_workers = (
             config.get(MAX_WORKERS_CONFIG_KEY) or DEFAULT_MAX_WORKERS)
 
-        self.dependencies = DependencySet()
-        for dep in prepare_dependencies(self):
-            self.dependencies.add(dep)
+        self.entrypoints = SpawningSet()
+        self.dependencies = SpawningSet()
+        self.subextensions = SpawningSet()
```

Comment:
I don't think it's possible to abstract away the differences. "Dependencies" can be injected into service code; they are the top level. Whereas "extensions" is the catch-all for everything managed by the container, including entrypoints.

We danced around having entrypoints and dependencies manage their own subextensions so the container didn't need to know about them. With more intelligent ``stop`` behaviour that may be be possible, but it currently isn't. If it does become possible then it'll be pretty easy to transition to it from here, and the ugly "subextensions" variable will go away.